src/linear_algebra/basic.lean

```diff
@@ -758,6 +758,18 @@ preserved under addition and scalar multiplication, then `p` holds for all eleme
   (H2 : ∀ (a:R) x, p x → p (a • x)) : p x :=
 (@span_le _ _ _ _ _ _ ⟨p, H0, H1, H2⟩).2 Hs h
 
+lemma span_eq_add_submonoid.closure {M : Type*} [add_comm_monoid M] (S : set M) :
+  (span ℕ S).to_add_submonoid = add_submonoid.closure S :=
+begin
+  refine (add_submonoid.closure_eq_of_le (by exact submodule.subset_span) _).symm,
```

Comment:
For some reason here `by exact submodule.subset_span` works, but `submodule.subset_span` doesn't.

Reply:
```suggestion
  refine (add_submonoid.closure_eq_of_le (submodule.subset_span : _) _).symm,
```
might work

Reply:
You might also want to use `submodule.to_add_submonoid_mono` here to explicitly rewrite the defeq `\le` statement.

Reply:
`submodule.subset_span : _` doesn't work (I also tried to give be even more explicit, without succeeding).

Are you sure `submodule.to_add_submonoid_mono` is relevant here? I only have one submodule, so I don't see how to use it.

Reply:
No, I'm not sure. What's the goal state inside this `by`?

Reply:
It's `S ⊆ ↑((span ℕ S).to_add_submonoid)`, and `subset_span` gives `S ⊆ ↑((span ℕ S)`.

Reply:
It looks like we're missing a `submodule.coe_to_add_submonoid` lemma. We already have `coe_to_add_subgroup`.

Reply:
I can add it, but it wouldn't change anything here, right? We can write `show... by ... rw` but it seems more complicated that the current `by exact`.

Reply:
I've added it in #7205. I agree, it's not so easy to use briefly here, what you have is fine.